### PR TITLE
Core toggle data prefix for popup attribute

### LINKED ISSUE #649: Core-toggle: popup-attribute will cause trouble from chromium 106-110+
Chrome (v104) logs an error in the developer console:
```
Found a 'popup' attribute. If you are testing the popup API, you must enable Experimental Web Platform Features. If not, note that custom attributes must start with 'data-': https://html.spec.whatwg.org/multipage/dom.html#custom-data-attribute. This usage will *likely cause site breakage* when the popup API ships: https://chromestatus.com/feature/5463833265045504. <nrkno-toggle class=​"kur-radio-toggle" popup hidden aria-labelledby=​"l6kho3wjqb2" role=​"group" id=​"l6kho3wj5oq">​</nrkno-toggle>​
```

Looks like Chrome is adopting the Open-UI proposal for pop-up API starting testing in v106-v110 with release planned for v110.
core-toggle `popup`-attribute needs to be renamed to `data-popup` to avoid conflicts in Chrome/Chromium

Resources:
- Open-UI pop-up API proposal: https://open-ui.org/components/popup.research.explainer
- Feature info on Chromestatus: https://chromestatus.com/feature/5463833265045504

### PR DESCRIPTION
**Pull request by skjalgepalg**
* Use `data-` prefix for popup attribute to avoid warnings and eventual conflict with the OpenUI pop-up API
* ~~Keep support for deprecated `popup` with warning to console to discourage use.~~
  * Removed support for `popup`, release will be a major bump

* Updated examples in docs to use `data-popup`
* Add tests for deprecation support
* Update existing test cases to use `data-popup`

Resolves #649

### CODE REVIEW

File: packages/core-toggle/core-toggle.js [hunk 1]
```diff
@@ -84,7 +84,16 @@ export default class CoreToggle extends HTMLElement {
   }
 
   // aria-haspopup triggers forms mode in JAWS, therefore store as custom attr
-  get popup () { return this.getAttribute('popup') === 'true' || this.getAttribute('popup') || this.hasAttribute('popup') }
+  get popup () {
+    const hasDeprecatedPopup = this.getAttribute('popup') === 'true' || this.getAttribute('popup') || this.hasAttribute('popup')
+    if (hasDeprecatedPopup) {
```
Comment (eirikbacker) [inline packages/core-toggle/core-toggle.js:89]: Could we rather have just `this.hasAttribute('popup')` as the `hasDeprecatedPopup` variable, and later return 
```
const key = hasDeprecatedPopup ? 'popup' : 'data-popup'
return this.getAttribute(key) === 'true' || this.getAttribute(key) || this.hasAttribute(key)
```
as this clarifies the purpose? :)

Comment (skjalgepalg) [reply packages/core-toggle/core-toggle.js:89]: Sure, that also feels less bloated. Stay tuned

**Review by frdrks — APPROVED** [review]
LGTM :shipit:

File: packages/core-toggle/core-toggle.js [hunk 2]
```diff
@@ -84,7 +84,17 @@ export default class CoreToggle extends HTMLElement {
...
+      console.warn(
+        this,
+        'uses deprecated `popup` attribute. Please use `data-popup` as specified in the docs (https://static.nrk.no/core-components/latest/index.html?core-toggle/readme.md). Note that old attribute name has been deprecated to avoid confusion with the `popup` attribute used by OpenUI pop-up API'
+      )
+    }
+    const key = hasDeprecatedPopup ? 'popup' : 'data-popup'
+    return this.getAttribute(key) === 'true' || this.getAttribute(key) || this.hasAttribute(key)
+  }
 
   set popup (val) { this[val === false ? 'removeAttribute' : 'setAttribute']('popup', val) }
```
Comment (laat) [inline packages/core-toggle/core-toggle.js:99]: burde vel også gjøre noe med attributten popup her.

**Comment by laat** [comment]
> * Keep support for deprecated `popup` with warning to console to discourage use.

why? It bloats and complicates. I think we should make a breaking change, and explain the data-popup change in the major release.

**Review by AndersCan — APPROVED** [review]

**Review by laat — APPROVED** [review]
Konge :)

**Pull request merged**